### PR TITLE
Adding salt key rotation guidance and link the hardening doc to the Salt best practices doc

### LINKED ISSUE #57725: [DOC] New doc on rotating keys
**Is your feature request related to a problem? Please describe.**
This works, but isn't explicit in the documentation, request for net new docs on key rotation

**Describe the solution you'd like**

- Rotating Keys
There are several reasons to rotate keys. One example is exposure or a compromised key. An easy way to rotate a key is to remove the existing keys and let the salt-master or salt-minion process generate new keys on restart.

**Rotate a minion key**
On the minion
salt-call saltutil.regen_keys
systemctl stop salt-minion

On the master
salt-key -d <minion-id>

On the minion
systemctl start salt-minion

On the master
salt-key -a <minion-id>

**Rotate a master key**

On the master
systemctl stop salt-master
rm <pki_dir>/master.{pem,pub}
systemctl start salt-master

On the minion
systemctl stop salt-minion
rm <pki_dir>/minion_master.pub
systemctl start salt-minion

**Rotate all keys**
https://github.com/dwoz/salt-rekey/

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by ScriptAutomate**
### What does this PR do?

- Adds `.. important::` directive message that helps the [Docs: Hardening Salt](https://docs.saltproject.io/en/latest/topics/hardening.html) and [Docs: Salt Best Practices](https://docs.saltproject.io/en/latest/topics/best_practices.html) link to each other.
- Adds Salt key rotation guidance to the [Docs: Hardening Salt](https://docs.saltproject.io/en/latest/topics/hardening.html)
- Adds links to these pages at the end of `SECURITY.md` for easy link guidance on securing Salt

### What issues does this PR fix or reference?

Fixes: #57725

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

**Comment by ScriptAutomate** [comment]
I've removed docs links to [salt-rekey](), due to that tool not being managed by the SaltStack org on GitHub, and for the disclaimer it is released under:

> _This script is provided without any fitness or warranty. It may cause minions to be disconnected from the Salt master and fail to reconnect, forcing manual intervention._

I have created the following feature request as a follow-up: #59342

**Review by Ch3LL — DISMISSED** [review]

File: SECURITY.md [hunk 1]
```diff
@@ -118,3 +118,12 @@ quickly and safely as is possible.
...
 mailing list. This list is low-traffic.
+
+## Guidance on Salt and security best practices
+
+Reference the following documentation to ensure Salt best practices are being
+implemented in your infrastructure:
+
+- [Docs: Hardening Salt](https://docs.saltproject.io/en/latest/topics/hardening.html)
+- [Docs: Salt Best Practices](https://docs.saltproject.io/en/latest/topics/best_practices.html)
+- [Blog: How I Hardened My Salt Environment](https://www.saltstack.com/blog/how-i-hardened-my-salt-environment/)
```
Comment (bryceml) [inline SECURITY.md:129]: is there a new url for this?

**Review by garethgreenaway — DISMISSED** [review]

File: SECURITY.md (hunk 1 above)
Comment (ScriptAutomate) [reply SECURITY.md:129]: Looks like that will be: https://saltproject.io/blog/how-i-hardened-my-salt-environment/
- We are currently working on getting this synced, and I will update upon confirmation

**Review by dwoz — DISMISSED** [review]

File: SECURITY.md (hunk 1 above)
Comment (ScriptAutomate) [inline SECURITY.md:129]:
```suggestion
- [Blog: How I Hardened My Salt Environment](https://saltproject.io/blog/how-i-hardened-my-salt-environment/)
```

**Review by garethgreenaway — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

File: SECURITY.md (hunk 1 above)
Comment (sagetherage) [reply SECURITY.md]: https://saltproject.io/security_announcements/how-i-hardened-my-salt-environment/

**Review by sagetherage — APPROVED** [review]

**Review by bryceml — APPROVED** [review]

**Pull request merged**